### PR TITLE
fix(push): send createDevice request only once per token change

### PR DESCRIPTION
**Pull request by Brazol**

### CODE REVIEW

File: packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart [hunk 1]
```diff
@@ -174,13 +178,24 @@ class StreamVideoPushNotificationManager implements PushNotificationManager {
       return;
     }
 
-    void registerDevice(String token, bool isVoIP) {
-      _client.createDevice(
+    void registerDevice(String token, bool isVoIP) async {
+      final tokenKey = isVoIP ? 'userDeviceTokenVoIP' : 'userDeviceToken';
```
Comment (deven98) [inline packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart:182]: Can we change this key to something unique and identifies this as a stream key? I feel like these are names that someone could use by accident in their own app.

File: packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart [hunk 2]
```diff
@@ -197,6 +212,11 @@ class StreamVideoPushNotificationManager implements PushNotificationManager {
             .listen((token) => registerDevice(token, true)));
   }
 
+  Future<void> removedStoredTokens() async {
+    await _sharedPreferences.remove('userDeviceToken');
```
Comment (deven98) [inline packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart:216]: Additionally can we store these keys as compile time constants somewhere else?

File: packages/stream_video_push_notification/pubspec.yaml [hunk 3]
```diff
@@ -24,7 +24,8 @@ dependencies:
   rxdart: ^0.28.0
   stream_video: ^0.5.5
   uuid: ^4.2.1
-
+  shared_preferences: ^2.3.2
+
```
Comment (deven98) [inline packages/stream_video_push_notification/pubspec.yaml:28]: Can you please add a changelog entry for everything?

**Review by deven98 — CHANGES_REQUESTED** [review]

File: packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart (hunk 1 above)
Comment (Brazol) [reply packages/stream_video_push_notification/lib/src/stream_video_push_notification.dart]: good point

**Pull request merged**